core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -574,7 +570,7 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
         case Empty =>
           process(None, isRawHttpAction)
 
-        case HttpEntity.Strict(ContentTypes.`application/json`, json) if !isRawHttpAction =>
+        case HttpEntity.Strict(ct, json) if WhiskWebActionsApi.isJsonFamily(ct.mediaType) && !isRawHttpAction =>
```

Comment:
How does this relate to the `customRejectionHandler` defined in `BasicHttpService`? I'm still unclear on the benefit of checking the content-type at all. Why not simply assume that all payloads are json regardless of content-type and return `HTTP 415 Unsupported Media Type` (with useful logging) in the event of a parse error?

Reply:
This is for the JSON part of web-actions specifically. We **need** to check that here, since we also support FormData etc.

This doesn't apply for the general API where we can do what you suggest.